dali/util/make_string.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.#ifndef DALI_MAKE_STRING_H
+
+#ifndef DALI_UTIL_MAKE_STRING_H_
+#define DALI_UTIL_MAKE_STRING_H_
+
+#include <string>
+
+namespace dali {
+
+namespace detail {
+
+template <class T>
+std::string make_string_delim(const std::string &delimiter, std::stringstream &ss, const T &val) {
+  ss << val;
+  return ss.str();
+}
+
+
+template <class T, class... Args>
+std::string make_string_delim(const std::string &delimiter, std::stringstream &ss, const T &val,
```

Comment:
This is bad design. It entangles two different concepts: print_delim and make_string_delim.
There should be a function print_delim taking a generic `std::ostream &` and make_string_delim written as:
```
template <typename Delimiter, typename... Args>
std::string make_string_delim(const Delimiter &delim, Args&& ...args) {
  std::stringstream ss;
  print_delim(ss, delim, std::forward<Args>(args)...);
  return ss.str();
}
```